framework/db/Query.php

```diff
@@ -844,6 +849,119 @@ public function addParams($params)
     }
 
     /**
+     * @var array aliases defined for tables used in this query. `tablename => alias`.
+     */
+    private $_aliases = [];
+
+    /**
+     * Populate table aliases used in this query.
+     *
+     * This function is used by [[from()]] and [[join()]] to make information about
+     * tables aliases available for [[getAlias()]].
+     *
+     * @param array $tables array of tables in the format allowed for [[from()]]:
+     *
+     * - numeric key, string value == table without alias, or table with alias specified after the name, separated by white space.
+     * - string key, string value == table with alias.
+     * - string key, obect value == subquery with alias. This alias will not be populated.
+     * @see from()
+     * @see join()
+     * @see getAlias()
+     * @since 2.0.7
+     */
+    protected function populateAliases($tables)
+    {
+        foreach ($tables as $alias => $tableName) {
+            if (is_object($tableName)) {
+                continue;
+            }
+            if (is_string($alias)) {
+                $this->_aliases[$tableName] = $alias;
+            } elseif (preg_match('/^(\\{\\{(\w+)\\}\\}|(.*?))(\s+AS\s+|\s+)(\\{\\{\w+\\}\\}|\w+)$/i', $tableName, $matches)) {
```

Comment:
It desn't look like all group results are used. Consider using `(?: something )`.